src/contexts/Products-Context.js

```diff
@@ -0,0 +1,78 @@
+import { createContext, useContext, useReducer } from "react";
+
+export const ProductsContext = createContext();
+
+export function ProductsProvider({ children }) {
+  const [state, dispatch] = useReducer(productsReducer, {
+    products: [],
+    sortBy: "none",
+    priceRange: 0
+  });
+
+  return (
+    <ProductsContext.Provider value={{ state, dispatch }}>
+      {children}
+    </ProductsContext.Provider>
+  );
+}
+
+function productsReducer(RedcState, action) {
+  switch (action.type) {
+    case "PRODUCTS":
+      return { ...RedcState, products: action.payload };
+      break;
+
+    case "SORT":
+      return { ...RedcState, sortBy: action.payload };
+      break;
+
+    case "PRICE_RANGE":
+      return { ...RedcState, priceRange: action.payload };
+      break;
+
+    case "RESET":
+      return { ...RedcState, sortBy: "none", priceRange: 0 };
+      break;
+
+    case "MICS":
+        console.log(action.isCheck)
+
+        console.log(action.isCheck.checked)
+      return {
+        ...RedcState,
+        products: RedcState.products.filter(
+          (item) => item.category === action.payload
+        )
+      };
+
+    default:
+      break;
+  }
+}
+
+export function useProductContext() {
+  return useContext(ProductsContext);
+}
+
+export function getSortedData(productsList, sortBy) {
```

Comment:
No Need to use this function inside context, you can use this inside your products page.

Reply:
Didn't strike that. Thanks for pointing out !